azure/scope/machinepool.go

```diff
@@ -405,8 +405,27 @@ func (m *MachinePoolScope) applyAzureMachinePoolMachines(ctx context.Context) er
 	}
 
 	existingMachinesByProviderID := make(map[string]infrav1exp.AzureMachinePoolMachine, len(ampms))
-	for _, machine := range ampms {
-		existingMachinesByProviderID[machine.Spec.ProviderID] = machine
+	for _, ampm := range ampms {
+		// propagate Machine delete annotation from owner machine to AzureMachinePoolMachine
+		// this ensures setting a deleteMachine annotation on the Machine has an effect on the AzureMachinePoolMachine
+		// and the deployment strategy.
+		machine, err := util.GetOwnerMachine(ctx, m.client, ampm.ObjectMeta)
+		if err != nil {
+			// TODO(mw): which verbosity? info or error?
```

Comment:
Consider adding something like

```
if apierrors.IsNotFound(err) {
	log.V(4).Info("...")
	continue
}
if err != nil {
	return errors.Wrap(err, "...")
}
```

Reply:
Alternatively, we could instead return with an error when it's not found, since we don't want to prematurely delete AzureMachines before their owner Machines come online, i.e. make sure that the MP controller has reconciled the Machines and so it's in a stable state.

Reply:
Logging the _Machine not found error_ as an `info` will preserve control flow as before. 
Do we really need to _return_ an error if owner Machine is not found? I don't think so.

Reply:
@nawazkh The idea is that if we can't find the Machine, it means that the CAPI MachinePool controller has not had time to create one yet, meaning the MachinePool scale up operation is not done. The delete machine annotation is added by the autoscaler, which only takes into account the Machines, not the AMPMs. So we want to wait for all the Machines to be created before proceeding with deletion since once the Machine is created, it might have the annotation added.

Reply:
I updated this part to return an error regardless of the type. I think what @Jont828 said makes sense. Let me know what you think!